File: Single_Number/Single_number.java
public class Single_number {
    public static void main(String[] args) {
        process(new int[] {2,2,1});
        process(new int[] {4,1,2,1,2});
        process(new int[] {1});
    }

    private static void process(int[] nums) {
        int single_number = 0;
        for (int num : nums) {
            single_number ^= num;
        }
        System.out.println(single_number);
    }
}

File: Single_Number/Single_number.py
def single_number(nums):
    unique_number = 0

    # XOR all numbers
    for num in nums:
        unique_number ^= num

    print("The unique number is:", unique_number)

single_number([2,2,1])
single_number([4,1,2,1,2])
single_number([1])     